src/docs/Core-2.0/ModuleInterfaces/MessageInterface.md

```diff
@@ -0,0 +1,41 @@
+Message Interface
+=================
+
+class:  `\Zikula\UsersModule\MessageModule\MessageModuleInterface`
+
+Modules that want the Core to identify the module as Message-capable must implement a class which
+implements this interface and then register that class in the container with the tag `zikula.message_module`.
+
+Please note the legacy module capability setting in `composer.json` is entirely disabled and will not work.
+
+The interface requires:
+
+    /**
+     * Get the url to a uid's inbox.
+     * If uid is undefined, use CurrentUserApi to check loggedIn status and obtain and use the current user's uid
```

Comment:
Better let the message module be responsible for it; this seems less error-prone because there are less locations where the check needs to be done.
In my opinion callers should not be required to make huge differences between normal users and the guest user.